src/sagas/reown.js

```diff
@@ -673,18 +803,24 @@ export function* onSessionProposal(action) {
   });
 
   if (reject) {
-    yield call(() => web3wallet.rejectSession({
-      id: params.id,
-      reason: {
-        code: ERROR_CODES.USER_REJECTED,
-        message: 'User rejected the session',
-      },
-    }));
+    try {
+      yield call(() => walletKit.rejectSession({
+        id,
+        reason: {
+          code: ERROR_CODES.USER_REJECTED,
+          message: 'User rejected the session',
+        },
+      }));
+    } catch (e) {
+      log.error('Error rejecting session on sessionProposal', e);
```

Comment:
question: The `walletKit` request on `onSessionProposal` below ( line 842 ) sends the error to Sentry if it happens, using `onExceptionCaptured()`.

Should this happen by default on all `walletKit` interactions?

Reply:
As we discussed, there is already an error handler upstream of here.